components/shared/ui/menu.tsx

```diff
@@ -1,53 +1,99 @@
 import Link from "next/link";
 import { FC, useState } from "react";
-import { mainColorBgClass } from "../../../lib/constants/colors";
+import { mainColorBgClass, mainColorBorderClass } from "../../../lib/constants/colors";
 import { useSiteCodename } from "../siteCodenameContext";
 import { Block_Navigation } from "../../../models";
 import { createItemSmartLink } from "../../../lib/utils/smartLinkUtils";
 import { ChevronDownIcon, Bars3Icon } from "@heroicons/react/24/solid";
+import { useRouter } from "next/router";
 
 type Link = Readonly<Block_Navigation>;
 
 type Props = Readonly<{
   item: Link;
 }>
 
+type MenuItemsProps = Readonly<{
+  items: Block_Navigation[];
+  activeMenu: string | number;
+  handleClick: (menuId: string | number) => void;
+}>
+
 type DropdownMenuProps = Readonly<{
   links: ReadonlyArray<Link>;
 }>;
 
-const DropdownMenu: FC<DropdownMenuProps> = props => {
+const MenuItems: FC<MenuItemsProps> = props => {
```

Comment:
I would create not menuItems as a list of `<'li> `elements but only MenuItem as one `<'li>`(instead of i`tems:Block_Navigation[]` it would take only `item: Block_navigation `and props.items.amp would in a component of one level higher). For me, it makes sense more semantically as it is a more atomic unit. And you can deal with state and variables of only one menuItem :)

Reply:
rewrtitten